### PR TITLE
8341440: ScrollPane: no immediate effect changing fitWidth/fitHeight

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
- fixed the issue
- added unit tests
- clarified default value of `ScrollPane.fitWidth` and `ScrollPane.fitHeight` properties

No CSR is needed as it is a minor clarification.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341440](https://bugs.openjdk.org/browse/JDK-8341440): ScrollPane: no immediate effect changing fitWidth/fitHeight (**Bug** - P4)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [John Hendrikx](https://openjdk.org/census#jhendrikx) (@hjohn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1597/head:pull/1597` \
`$ git checkout pull/1597`

Update a local copy of the PR: \
`$ git checkout pull/1597` \
`$ git pull https://git.openjdk.org/jfx.git pull/1597/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1597`

View PR using the GUI difftool: \
`$ git pr show -t 1597`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1597.diff">https://git.openjdk.org/jfx/pull/1597.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1597#issuecomment-2405763899)

### CODE REVIEW

File: modules/javafx.controls/src/main/java/javafx/scene/control/ScrollPane.java [hunk 1]
```diff
@@ -354,6 +354,8 @@ public final DoubleProperty vmaxProperty() {
      * If true and if the contained node is a Resizable, then the node will be
      * kept resized to match the width of the ScrollPane's viewport. If the
      * contained node is not a Resizable, this value is ignored.
+     *
+     * @defaultValue false
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/main/java/javafx/scene/control/ScrollPane.java:358]: This is unrelated to the fix, and you were otherwise not touching this file. Why change it?

**Review by kevinrushforth — COMMENTED** [review]
Fix and test both look good. I confirm that the newly added test and the program attached to the JBS issue pass with the fix and faill without it.

Not sure why you modified the docs, though. It seems unrelated.

File: modules/javafx.controls/src/main/java/javafx/scene/control/ScrollPane.java (hunk 1 above)
Comment (andy-goryachev-oracle) [reply modules/javafx.controls/src/main/java/javafx/scene/control/ScrollPane.java:358]: it is related to the fix, may be we differ in the _degree_ of relation.  I did not want to create a separate ticket for such a minor clarification...  but maybe I should, and fix all the missing `@defaultValue`s across the controls.

Comment (andy-goryachev-oracle) [reply modules/javafx.controls/src/main/java/javafx/scene/control/ScrollPane.java:358]: Filed https://bugs.openjdk.org/browse/JDK-8341925

**Review by hjohn — APPROVED** [review]
Looks okay.

minor: calling `viewContent.requestLayout()` suffices. The other two calls are superfluous.

**Review by kevinrushforth — APPROVED** [review]
Looks good.
